### ISSUE TITLE
[$500] Public room - App crashes when open public room via deeplink

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#48287]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.26-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4904922&group_by=cases:section_id&group_order=asc&group_id=229066
**Email or phone of affected tester (no customers):** gocemate+a2019@gmail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. While logged out from android app, via deep link navigate to this public room link: https://staging.new.expensify.com/r/2091104345528462

2. When public room opens, via deep link navigate to this public room link: https://staging.new.expensify.com/r/6776919265133947

## Expected Result:
Link should open in app

## Actual Result:
App crashes when open public room via deep link while other public room is open

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/474571ac-6e74-49dc-9da8-5a1755e49d0d

[logs (2).txt](https://github.com/user-attachments/files/16803570/logs.2.txt)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021831388703029188039</li>
        <li>Upwork Job ID: 1831388703029188039</li>
        <li>Last Price Increase: 2024-10-08</li>
    </ul>
</details>

**Comment by lanitochka17** [comment]
@kevinksullivan  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by mkzie2** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
App crashes when open public room via deep link while other public room is open

### What is the root cause of that problem?

This is the crash error log:

![Screenshot 2024-08-30 at 01 47 21](https://github.com/user-attachments/assets/e24bffe6-feeb-413c-8980-97b3a8d29d8a)

We fallback `reportID` to empty string:

https://github.com/Expensify/App/blob/15805221bdd48e88bc39991895cbb26172bb106e/src/pages/home/ReportScreen.tsx#L172

But when using in `useOnyx`'s keys, we only fallback to `-1` when the `reportID` is `undefined` or `null` because we use nullish coalescing `??`:

https://github.com/Expensify/App/blob/ddbe832a84136be27043cd518d4af5fc2dfe4021/src/pages/home/report/ReportActionItem.tsx#L191

### What changes do you think we should make in order to solve the problem?


 Use logical or instead `||`.

**Comment by rushatgabhane** [comment]
hmmm

**Comment by rushatgabhane** [comment]
> We should check all likewise usages and use logical OR instead ||

@mkzie2 can you please correct this. Do you mean use `||` instead of `??`

**Comment by mkzie2** [comment]
@rushatgabhane Yes that's correct, but only when the `reportID` might be empty string,

**Comment by rushatgabhane** [comment]
@mkzie2 earlier when we did that, we encountered a bug- 
https://expensify.enterprise.slack.com/archives/C01GTK53T8Q/p1725902444282339?thread_ts=1725890512.566159&cid=C01GTK53T8Q


can you please see if your suggestion will not cause it?

**Comment by mkzie2** [comment]
@rushatgabhane Can you summarize its content because I haven't had access to Slack yet? Thanks.

**Comment by rushatgabhane** [comment]
1. manually clear report_[report ID] from onyx
2. navigate to the report - r/reportID
3.  OpenReport command is not called in network tab
4. So the report keeps loading forever

**Comment by kevinksullivan** [comment]
Still waiting on proposals. Looping in another BZ member as I'm going OOO

**Comment by rushatgabhane** [comment]
waiting for proposals

**Comment by wildan-m** [comment]
@rushatgabhane I can't consistently reproduce it, but it happened once, do you know steps with more consistent result?

**Comment by kevinksullivan** [comment]
Still waiting on proposals

**Comment by kadiealexander** [comment]
Still looking for proposals, I don't think this is high enough priority to raise the price right now.

**Comment by kadiealexander** [comment]
Still looking for proposals, I don't think this is high enough priority to raise the price right now.

**Comment by kadiealexander** [comment]
Requested a retest [here](https://github.com/Expensify/App/issues/48287).

**Comment by kavimuru** [comment]
Bug is reproducible.

https://github.com/user-attachments/assets/f4816b40-0f37-48f4-83dd-3ac79a1098d6

**Comment by kadiealexander** [comment]
Alright I think we can bump the price on this one.

**Comment by CyberAndrii** [comment]
Is it still crashing? It doesn't seem to be based on [the video above](https://github.com/Expensify/App/issues/48287#issuecomment-2399872671) :thinking: :thinking: :thinking:. cc @kavimuru

Anyway, I think @mkzie2's [proposal](https://github.com/Expensify/App/issues/48287#issuecomment-2318623880) is correct and the same fix as proposed was implemented in https://github.com/Expensify/App/pull/48962. For some reason, it was later reverted in https://github.com/Expensify/App/pull/49542 and then re-added in https://github.com/Expensify/App/pull/49255 (deployed to prod on Monday).

**Comment by NJ-2020** [comment]
Issue is not reproducible

**Comment by kadiealexander** [comment]
Alright, closing.
